core/crypto/client_impl.go

```diff
@@ -44,31 +46,39 @@ func (client *clientImpl) NewChaincodeDeployTransaction(chaincodeDeploymentSpec
 	}
 
 	// Get next available (not yet used) transaction certificate
-	tCert, err := client.tCertPool.GetNextTCert()
+	tCerts, err := client.tCertPool.GetNextTCerts(1)
 	if err != nil {
 		client.error("Failed getting next transaction certificate [%s].", err.Error())
 		return nil, err
 	}
+	if len(tCerts) != 1 {
+		client.error("Failed obtaining a TCert for Chaincode execution [%s].", err.Error())
```

Comment:
Thank you @angrbrd. Good point, these messages should be consistent. While at it, I actually updated them all (the new and the previous) to have the more correct form of 'Failed to obtain %d TCerts ....'. Please could you take another look at [3f43948](https://github.com/hyperledger/fabric/pull/1714/commits/3f43948b10e50a560ecd852b7ca09e46e8e1ceb2) to verify that this makes sense/I haven't missed anything?